### ISSUE TITLE
Issues loading places script for SearchField 

### ISSUE THREAD

**Issue by MidnightP** [istarkov/google-map-react#303]
In the container my page component I'm loading the places script including my key and language. 

```
const key = process.env.GOOGLE_API_KEY
    const language = 'NL'

    const placesUrl = `https://maps.google.com/maps/api/js?key=${ key }&language=${ language }libraries=places&sensor=false`

    load(placesUrl, { async: true }, (err, script) => {
      if (err) { console.log('error in load-script: ', err) }
      else { console.log('succes in load-script script.src: ', script.src) }
    })
```

Then in a child container I'm using the ```<GoogleMap />```-tag feeding it only an options object and defaultCenter and defaultZoom. In my console I receive the following error:

> You have included the Google Maps API multiple times on this page. This may cause unexpected errors.
> Google Maps API warning: NoApiKeys
> Google Maps API warning: SensorNotRequired

Also, I recieve an error for the searchField itself, copied from the searchbox example in readme:

> Uncaught ReferenceError: google is not defined

Can I feed options and defaults in GoogleMap component, even after the script was already loaded? Is there a working example with the searchField included ?

**Comment by MidnightP** [comment]
The sensor parameter is not required anymore I read

**Comment by istarkov** [comment]
Have u solved the problem, I know it and can help if needed.

**Comment by smirea** [comment]
@istarkov do you know how to solve the double-loading issue? I'm trying to use this package together with `google-maps` for when I want to use places autocomplete, but neither library detects each-other. Do you know by any chance how to disable API loading on this gmaps?

**Comment by istarkov** [comment]
The only way is to use custom loader which you can provide via `googleMapLoader` property,
it must return `Promise` of `window.google.maps` which google api defines at load.

Internally if nothing provide is used this loader:
https://github.com/istarkov/google-map-react/blob/master/src/utils/loaders/google_map_loader.js

So the idea is to return Promise which will be resolved only after `third-party` or your own google-map api will be loaded.

**Comment by matmac** [comment]
Is this problem solved? I have the same issue, can use places autocomplete...

**Comment by istarkov** [comment]
@matmac the answer is directly above your comment,
however if you just want to use places lib in other place without map

```javascript
// bootstrapURLKeys.js
export const bootstrapURLKeys = {
  ...yourKeys,
  libraries: ['places'].join(','),
}

// places service.js 
import GoogleMapReact from 'google-map-react';
const { googleMapLoader } = GoogleMapReact;
import { bootstrapURLKeys } from './bootstrapURLKeys';
...
googleMapLoader(bootstrapURLKeys)
        .then((maps) => {
          const places = new maps.places.PlacesService(fakeElt);
          // use places
        }
...

// map.js
import { bootstrapURLKeys } from './bootstrapURLKeys';
<GoogleMap bootstrapURLKeys={bootstrapURLKeys}>
```

**Comment by matmac** [comment]
@istarkov I need to use the places feature to change the map since I'm a React newbie I don't think I can put together a custom loader. Anyways I'll find the way, but nevertheless, it would be nice to have this feature included.

**Comment by sonsoleslp** [comment]
Is there any documentaton on the googleMapLoader prop? Can I pass an already loaded map instance?

**Comment by itsmichaeldiego** [comment]
@sonsoleslp Sorry for the late response, but no, you can use the maps instance that we create after the map is loaded, [check this example](https://github.com/istarkov/google-map-react/issues/512#issuecomment-366352763).

About the instances getting mixed up, that did not happen to me, would you mind creating a Codepen and I can take a look at it? 

@matmac If your problem still persists, would you mind creating a new issue? I can take a look at it thoroughly if that is the case.

**Comment by lukas-alliado** [comment]
As I did not really understand any of the answers and examples completely above I'll share my solution on how to use this with ` 'react-places-autocomplete'`

```tsx
import PlacesAutocomplete from 'react-places-autocomplete';
import { getMapKeys } from './map/mapUtils';

const googleMapLoader = require('google-map-react/lib/loaders/google_map_loader').default;

class _GoogleMapsAutoCompleteServiceBased extends React.Component {
  constructor(props: PropsWithStyles) {
    super(props);
    this.state = {
      mounted: false,
    };
  }

  public componentDidMount(): void {
    // If google maps is not present yet, load it via 
    // googleMapLoader from google-map-react to avoid loading the api twice
    if (!window.google) {
      googleMapLoader(getMapKeys())
        .then((maps: any) => {
      // make api available globaly
          window.google = { maps };
          this.setState({ mounted: true });
        })
        .catch((error: any) => {
          logService.error('AutoComplete', error);
          globalMessageService.pushMessage({
            message: locale.common.errors.autoCompleteError,
            variant: 'error',
          });
        });
    } else {
      this.setState({ mounted: true });
    }
  }
  render() {
    if (this.state.mounted === false) {
      return <Loading/>;
    }

    // maps api is now globally available and can be used by `PlacesAutocomplete`
    return <PlacesAutocomplete ..../>
}
```

This ensures that the Autocomplete works even if the map is not present on the same page, but the map will reuse the same instance loaded here.

If the map was loaded before the autocomplete it will just reuse the existing instance.


This behaviour should definitely be documented clearer ...

**Comment by sunviwo** [comment]
Hi @lukas-alliado, thanks for your solution. But I can't see where you use the map instance in your script. Could you give further explanations please? Thanks in advance.

**Comment by lukas-alliado** [comment]
> Hi @lukas-alliado, thanks for your solution. But I can't see where you use the map instance in your script. Could you give further explanations please? Thanks in advance.

sorry for the late reply.

I updated my snippet with a more detailed explanation for future references:
The code make the Maps API available globally on ´window.google.maps´
